java/ql/test/kotlin/query-tests/ConstantLoopCondition/A.kt

```diff
@@ -0,0 +1,24 @@
+fun fn0(f: Function0<Unit>) = f()
+
+fun fn1() {
+    var c = true
+    while (c) {
+        fn0 {
+            c = false
```

Comment:
So this is legal Kotlin? That's definitely not legal in Java. How are captured local variables compiled in Kotlin? This seems to indicate that our current handling of captured variables is completely wrong when applied to Kotlin.

Reply:
We may need two different concepts to describe the two different kinds of variable capture, and we should be careful with adding general purpose predicates that falsely indicates that there's only one such concept.